File: collector/src/domain/weather.py
from dataclasses import dataclass, asdict
from datetime import datetime

@dataclass
class WeatherData:
    location_lat: float
    location_lon: float
    temperature: float
    humidity: float
    wind_speed: float
    condition_code: int
    # Cria o timestamp automaticamente se não for passado
    collected_at: str = None 

    def __post_init__(self):
        if self.collected_at is None:
            self.collected_at = datetime.now().isoformat()

    def to_dict(self):
        """Converte a classe para dicionário (JSON)"""
        return asdict(self)

File: worker/internal/domain/weather.go
package domain

type WeatherData struct {
	// Ajustado aqui para ler o JSON plano que vem do Python
	LocationLat float64 `json:"location_lat"`
	LocationLon float64 `json:"location_lon"`

	Temperature   float64 `json:"temperature"`
	Humidity      float64 `json:"humidity"`
	WindSpeed     float64 `json:"wind_speed"`
	ConditionCode int     `json:"condition_code"`
	CollectedAt   string  `json:"collected_at"`
}